### DISCUSSION TITLE
Disable private channels

### DISCUSSION THREAD

**Discussion by MichaelAntonFischer** [lightningnetwork/lnd#8743] [Q&A] [answered]
If I read the documentation right, private channels should be off by default on 17.4, but they are accepted.

How can I block private channels?

**Comment by ellemouton** [comment]
Private channels are off by default if you are the channel creator. Ie, if you open a channel, it will by default be non-private (un-announced rather). If you want to stop peers from opening un-announced channels to you, then you can write a script that uses the `ChannelAcceptor` endpoint to intercept any channel open requests - then you can inspect the `ChannelFlags` of the request and reject any request where the flags do not contain the "Announce" funding bit

**Reply by MichaelAntonFischer** [reply]
Has anyone already done that?

**Reply by bitromortac** [reply]
You could try https://github.com/callebtc/electronwall, which has an example of how to act on private channel opens: https://github.com/callebtc/electronwall/blob/main/rules/ChannelAccept.js (I haven't used this myself).

**Reply by blckbx** [reply, marked as answer]
Also BoS: https://github.com/alexbosworth/balanceofsatoshis#inbound-channel-rules 

Run as a service like:
```
bos inbound-channel-rules --rule "NOT(PRIVATE)"
```

**Reply by MichaelAntonFischer** [reply]
Thanks for the bos tip. It was a bit of a hassle to get it running in a containerized setup, but it works perfectly.
